plugins/fetcher.cc

```diff
@@ -113,9 +112,16 @@ struct FetcherFunction: public ValueFunctionT<FetcherArgs, FetcherOutput> {
                 blob = CellValue::blob(streamo.str());
             }
 
-            result.content = ExpressionValue(std::move(blob), info.lastModified);
+            FsObjectInfo info = stream.info();
```

Comment:
OK, so... sorry to do this again... the filter_streams will throw their exceptions when close() is called.  So whenever we want to make sure that an exception is thrown, we need to call close().  Why?  Because sometimes the exception happens in a worker thread, which will crash the exception if it's thrown.  Can we try to make close() throw the exception, in other words complete any missing logic in the filter_stream::close() method rather than putting it here?